CleanWeather/Scenes/FavouriteCities/FavouriteCitiesController.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  FavouriteCitiesController.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 04/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import UIKit
+
+protocol FavouriteCitiesPresentable: SpinnerPresentable & AlertPresentable {
+    func displayCities(_ city: [FavouriteCitiesListDisplayable])
+}
+
+final class FavouriteCitiesController: SharedViewController {
+
+    @IBOutlet private weak var citiesTableView: UITableView!
+
+    private let interactor: FavouriteCitiesInteractor
+
+    private var citiesDataSource = [FavouriteCitiesListDisplayable]()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setupNavigationBar()
+        interactor.getCities()
+    }
+
+    init(interactor: FavouriteCitiesInteractor) {
+        self.interactor = interactor
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    deinit {
+        print("FavouriteCitiesController DEALLOCATED")
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension FavouriteCitiesController: FavouriteCitiesPresentable {
+
+    func displayCities(_ city: [FavouriteCitiesListDisplayable]) {
+        citiesDataSource = city
+        citiesTableView.reloadData()
+    }
+}
+
+extension FavouriteCitiesController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return citiesDataSource.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeue(with: FavouriteCitiesTableViewCell.self, for: indexPath)
+        cell.setupFavouriteCitiesCell(with: citiesDataSource[indexPath.row])
```

Comment:
ta cellka to:
FavouriteCitiesTableViewCell

czy myslisz ze ma sens nazywac jej publiczna metode:
cell.setupFavouriteCitiesCell

nazwij:
cell.setup(with city: Displayable...)